registry/txt.go

```diff
@@ -194,11 +194,13 @@ func (im *TXTRegistry) generateTXTRecord(r *endpoint.Endpoint) []*endpoint.Endpo
 	// old TXT record format
 	txt := endpoint.NewEndpoint(im.mapper.toTXTName(r.DNSName), endpoint.RecordTypeTXT, r.Labels.Serialize(true)).WithSetIdentifier(r.SetIdentifier)
 	txt.ProviderSpecific = r.ProviderSpecific
+	txt.Labels[endpoint.OwnedRecordLabelKey] = r.DNSName
```

Comment:
This looks a bit misplaced. The TXT registry is shared by all providers, yet `endpoint.OwnedRecordLabelKey` is only used in the AWS provider. I can imagine that this could apply to all other providers as well but maybe it's better to make it part of `r.ProviderSpecific` (one line before) such that we don't need this line.